### ISSUE TITLE
训练后数据的独特点少于类别数，从而影响准确率

### ISSUE THREAD

**Issue by pikacai119** [XLearning-SCU/2021-CVPR-Completer#23]
作者您好,
我在直接将代码以zip方式下载并解压缩后，首先修改了损失值的计算方式为pytorch <= 1.2（默认的计算对应的是 pytorch > 1.2），使用的环境如下：
pytorch==1.2.0 

numpy==1.19.1

scikit-learn==0.23.2

munkres==1.1.4

但是在执行完脚本命令python run.py --dataset 0 --devices 0 --print_num 100 --test_time 5
之后，会有如下警告信息：
2021-CVPR-Completer-main\evaluation.py:136: ConvergenceWarning: Number of distinct clusters (3) found smaller than n_clusters (20). Possibly due to duplicate points in X. cluster_obj.fit(x)
也就是数据的独特点（distinct points）数量少于指定的簇数20（即只有 3 个独特簇）。

并且会导致聚类的acc大幅下降，得到的相关结果为：
2025-07-01 22:33:07 - root - INFO: - view_concat {'kmeans': {'AMI': -0.002, 'NMI': 0.0028, 'ARI': 0.0027, 'accuracy': 0.2334, 'precision': 0.0294, 'recall': 0.0488, 'f_measure': 0.0361}}
2025-07-01 22:33:26 - root - INFO: - Epoch : 500/500 ===> Reconstruction loss = 0.6991===> Reconstruction loss = 0.0622 ===> Dual prediction loss = 0.0000  ===> Contrastive loss = -4.3667e+02 ===> Loss = -4.3660e+02
2025-07-01 22:33:27 - root - INFO: - --------------------Training over--------------------
2025-07-01 22:33:27 - root - INFO: - ACC:[0.2293, 0.2376, 0.2439, 0.2326, 0.2334]
2025-07-01 22:33:27 - root - INFO: - NMI:[0.0057, 0.0054, 0.0045, 0.0049, 0.0028]
2025-07-01 22:33:27 - root - INFO: - ARI:[-0.0053, -0.0043, 0.0038, -0.0014, 0.0027]
2025-07-01 22:33:27 - root - INFO: -  ACC 23.54 std 0.50 NMI 0.47 std 0.10 ARI -0.09 std 0.36

请问您有遇到过类似问题吗，或者我上述的步骤有问题吗，应该如何解决呢？

**Comment by pikacai119** [comment]
@Lin-Yijie 望您有时间查看一下

**Comment by Lin-Yijie** [comment]
感谢关注！没遇到过这种问题。看起来可能问题如下 1）模型塌掉了,表征倾向于identity，导致kmeans聚类出现问题。2）kmeans那边设置有误。

如果没有修改其他代码的话，可能是loss优化出了问题。
